battlesim.py

```diff
@@ -65,17 +92,23 @@ def get_total_dps(self):
                 dps += ship.dps
         return dps
 
+    def get_attacked_ship(self):
+        for ship in self.ships:
+            if ship.isAlive():
+                return ship
+        return None
+
     def get_total_hps(self):
         hps = 0
         healer_being_attacked = True
         for ship in self.ships:
             if ship.isAlive():
-                if self.hasTanker() and self.hasHealer():
+                if self.hasTanker() and self.hasHealer() and (self.get_attacked_ship().isTanker() == False):
```

Comment:
self.hasHealer() is redundant here. If fleet has tanker and is not being attacked is good enough for mid_slot * 2